### PR TITLE
Added functionality for the event history

### PR DESCRIPTION
**Pull request by maurhofer-ubique**

### CODE REVIEW

File: dp3t-sdk/sdk/src/main/java/org/dpppt/android/sdk/internal/SyncWorker.java [hunk 1]
```diff
@@ -255,11 +279,21 @@ private static long getLastDesiredSyncTime(DayDate dateToLoad) {
...
+			return true;
+		}
+	}
+
 	private static void uploadPendingKeys(Context context) {
 		AppConfigManager appConfigManager = AppConfigManager.getInstance(context);
 		PendingKeyUploadStorage pendingKeyUploadStorage = PendingKeyUploadStorage.getInstance(context);
 		PendingKeyUploadStorage.PendingKey pendingKey = null;
 		try {
+			int numPendingUploaded = 0;
```
Comment (simonroesch) [inline dp3t-sdk/sdk/src/main/java/org/dpppt/android/sdk/internal/SyncWorker.java:296]: this is never increased and will always be 0

Comment (simonroesch) [reply dp3t-sdk/sdk/src/main/java/org/dpppt/android/sdk/internal/SyncWorker.java:296]: also we should differentiate between fake uploads and normal ones

File: dp3t-sdk/sdk/src/main/java/org/dpppt/android/sdk/internal/history/HistoryDatabase.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	}
+
+	public void addEntry(HistoryEntry entry) {
+		SQLiteDatabase db = dbHelper.getWritableDatabase();
+		insert(db, entry);
+	}
+
+	protected static void insert(SQLiteDatabase db, HistoryEntry entry) {
+		ContentValues values = new ContentValues();
+		values.put(HistorySpec.COLUMN_NAME_TYPE, entry.getType().ordinal());
```
Comment (simonroesch) [inline dp3t-sdk/sdk/src/main/java/org/dpppt/android/sdk/internal/history/HistoryDatabase.java:50]: This is very dangerous, if we refactor the enum, the history gets fucked up!

File: dp3t-sdk/sdk/src/androidTest/java/org/dpppt/android/sdk/internal/history/HistoryDbTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		ArrayList<HistoryEntry> entries = new ArrayList<>();
+		entries.add(new HistoryEntry(HistoryEntryType.SYNC, "AAB", true, 1591806623068L));
+		entries.add(new HistoryEntry(HistoryEntryType.OPEN_APP, null, true, 1591855343112L));
+		entries.add(new HistoryEntry(HistoryEntryType.OPEN_APP, null, true, 1591855355203L));
+		for (HistoryEntry entry : entries) { db.addEntry(entry); }
+
+		db.clear();
+
+		List<HistoryEntry> addedEntries = db.getEntries();
+		assert 0 == addedEntries.size();
```
Comment (buenzli-ubique) [inline dp3t-sdk/sdk/src/androidTest/java/org/dpppt/android/sdk/internal/history/HistoryDbTest.java:38]: you have to use the JUnit assert methods

**Pull request merged**